### DISCUSSION TITLE
login configuration

### DISCUSSION THREAD

**Discussion by ways2learning** [cypress-io/cypress-realworld-app#1027] [Q&A]
Hi @amirrustam @kevinold @davidkpiano, I am not from a programming background and trying to understand the UI tests code of this application. But, one thing where I have been stuck up and trying to figure out was the configuration of the test for ‘cy.login(user.username)’ command. I have gone through the entire application code but, unable to understand how does the find command is always picking up the first user as ‘Katharina_Bernier’ from the seeded database and where it has been configured in such a way and how can we change it if we need to log in for any other user? I request you to please help me in understanding this. Your answering on this would help me a lot and be appreciated. Thank you.

**Comment by kevinold** [comment]
@ways2learning 

[This test is an example](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/tests/ui/auth.spec.ts#L26-L31) of what you describe in where we use the `cy.database` command with `find` and it will get the *first* user in our database.

The `find` is implemented inside of this [Cypress task, `find:database`](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/plugins/index.ts#L65-L67), using lodash's `find` method to take the first element from the [array returned from list of users returned from the api request](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/plugins/index.ts#L44-L51)

> how can we change it if we need to log in for any other user?

To login as another user, pass `filter` to `cy.database`, [as demonstrated in this beforeEach](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/tests/ui/transaction-feeds.spec.ts#L54-L59) where you can select any set of users from the array returned.

**Reply by ways2learning** [reply]
Glad to see your quick reply on this Kevin. 
It is working and I am able to login as another user after passing 'filter' method to cy.database().

Thanks a lot once again.
